### PR TITLE
[RFR] Implement Headroom in AppBar

### LINKED ISSUE #1819: Two horizontal scroll in version ^2.0.0-RC3
Hello, I'm using the react-admin version ^2.0.0-RC3, but I get an error on a table with multiple columns with 2 horizontal scroll (in body and table), the following image is my error:

img

I want only horizontal scroll in table and columns is not stretch, so can you support with options in <Resource>

Thank you.

### PR DESCRIPTION
**Pull request by zyhou**
## Description

We have some issue on AppBar component like #1819. 

In fact, we don't follow [ the material design guidelines](https://material.io/design/components/app-bars-top.html#behavior), which specify that the scrollbar should hide on scroll down, and show up (fixed) on scroll up (see https://material-components.github.io/material-components-web-catalog/#/component/top-app-bar/standard for an example).

A discussion on the same subject in the [material-ui tracker: mui-org/material-ui#12337](https://github.com/mui-org/material-ui/issues/12337).

## Solution

I implement [react-headroom](https://github.com/KyleAMathews/react-headroom) container in AppBar component. I export `<Headroom />` component which is already configure for react-admin.

- [x] Clean example
- [x] Documentation
- [x] Add test E2E
- [x] Test Chrome
- [x] Test Firefox
- [x] Test Edge
- [x] Test IE

## Result

peek 2018-08-24 12-11
peek 2018-08-24 12-10

fix #1819

### CODE REVIEW

File: cypress/integration/list.js [hunk 1]
```diff
@@ -15,6 +15,22 @@ describe('List Page', () => {
...
+            cy.get(ListPagePosts.elements.title).contains('Posts List');
+        });
+
+        it('should hide the appBar when scroll down action appears', () => {
+            cy.scrollTo(0, 500);
+            cy.get(ListPagePosts.elements.headroom).should('not.be.visible');
+        });
+
+        it('should show the appBar when scroll top action appears', () => {
+            cy.scrollTo(0, -500);
```
Comment (djhi) [inline cypress/integration/list.js]: Wouldn't it be better to only scroll up a little (less than 500, maybe only 200) ?

Comment (djhi) [reply cypress/integration/list.js]: My understanding is that it should appear when you start scrolling up, right ?

Comment (zyhou) [reply cypress/integration/list.js]: - Hep, I can 200px or less (like 100px), I try in local

- Yes exactly

File: docs/_layouts/default.html [hunk 2]
```diff
@@ -634,6 +634,9 @@
                             <li class="chapter">
                                 <a href="#using-a-custom-layout">Using a Custom Layout</a>
                             </li>
+                            <li class="chapter">
+                                <a href="#using-a-custom-appbar">Using a Custom AppBar</a>
```
Comment (djhi) [inline docs/_layouts/default.html]: I think this should be a section of the `Using a Custom Layout` chapter

Comment (djhi) [reply docs/_layouts/default.html]: Nevermind this one

File: cypress/integration/list.js [hunk 3]
```diff
@@ -15,6 +15,18 @@ describe('List Page', () => {
         });
     });
 
+    describe('Headroom', () => {
```
Comment (fzaninotto) [inline cypress/integration/list.js]: s/Headroom/Auto-hide AppBar/

**Pull request merged**